tests/cpptests/test_cpp_thpool.cpp

```diff
@@ -173,20 +173,16 @@ TEST_P(PriorityThpoolTestFunctionality, TestTerminateWhenEmpty) {
     redisearch_thpool_wait(this->pool);
 
     // Wait for the threads to terminate.
-    thpool_stats stats = redisearch_thpool_get_stats(this->pool);
-    while (stats.num_threads_alive) {
+    while (redisearch_thpool_get_stats(this->pool).num_threads_alive) {
```

Comment:
The issue with the test was that the expression within the while was not volatile?